backend/layers/common/entities.py

```diff
@@ -181,6 +180,7 @@ class DatasetVersion:
     metadata: DatasetMetadata
     artifacts: List[DatasetArtifact]
     created_at: datetime
+    revised_at: Optional[datetime]  # The last time this Dataset Version was Published
```

Comment:
If we put `revised_at` here, this is functionally equivalent to `published_at` since a collection version can only be published once. If you want to get `revised_at` for a _canonical_ collection, we have two options:
1. Save the field to the canonical table similarly to `originally_published_at`, and update it each time we publish a version (revision)
2. Infer it as the `published_at` of the most recent version

Reply:
So there are two timestamp we care about in a dataset revision.
1. `dataset.published_at` which is the timestamp when the dataset was **first** published, which may not be the same as `collection.published_at` if the dataset was added after the collection was first published.
2. `dataset.revised_at` which is the timestamp when a revision for a dataset was last published. If the dataset were ever changed/reuploaded, this value would be different from both `collection.published_at` and `dataset.published_at`.

I didn't see a value in the database table to represent `revised_at`